### PR TITLE
New Types

### PR DESCRIPTION
**Pull request by iammoonman**
Adds layouts and fields consistent with the Scryfall API as of Jan 2024.

### CODE REVIEW

File: card.go [hunk 1]
```diff
@@ -585,6 +617,8 @@ type CardFace struct {
 	// This will only be set if the card is not in English.
 	PrintedName *string `json:"printed_name"`
 
+	FlavorName *string `json:"flavor_name,omitempty"`
```
Comment (BlueMonday) [inline card.go:620]: I was looking through the documented card face object and I don't see this one there. I only see it on the card object. Is the documentation incorrect? Do you happen to have an example card that has a card face with a flavor name?

https://scryfall.com/docs/api/cards#card-face-objects

**Comment by BlueMonday** [comment]
Thank you for helping keep the library up to date! Just have one question about one of the new fields!

File: card.go (hunk 1 above)
Comment (iammoonman) [reply card.go:620]: https://scryfall.com/card/iko/378/brokkos-apex-of-forever

Comment (BlueMonday) [reply card.go:620]: Hmmm that one seems to not have multiple card faces so it only has the flavor_name on the card object and not on the card face object.

**Comment by BlueMonday** [comment]
Made a few minor changes, hope you don't mind: https://github.com/BlueMonday/go-scryfall/commit/e6cfda2e7c044e25e82d7f9df3547f4fe34def0d

Also once again thank you for your contribution and helping keep the library up to date!

**Comment by BlueMonday** [comment]
Version [v0.4.0](https://github.com/BlueMonday/go-scryfall/releases/tag/v0.4.0) of this library should contain your changes.

**Comment by iammoonman** [comment]
Thanks!

**Pull request merged**